spec/models/zone_spec.rb

```diff
@@ -192,6 +192,28 @@
     end
   end
 
+  context "maintenance zone" do
+    it "creates missing region during seed" do
+      expect(MiqRegion.my_region).to be_nil
+      described_class.seed
+      expect(MiqRegion.my_region).to be_present
+    end
+
+    it "is seeded with relation to region" do
+      described_class.seed
+      zone = Zone.where("name LIKE ?", "#{Zone::MAINTENANCE_ZONE_NAME_PREFIX}%").first
+
+      expect(described_class.maintenance_zone).to be_present
+      expect(described_class.maintenance_zone.id).to eq(zone.id)
+      expect(MiqRegion.my_region&.maintenance_zone).to eq(zone)
```

Comment:
The lonely operator shouldn't be necessary here.  The test may need to seed the region first.

Reply:
I don't understand this philosophy of "expect some value there because I'm pretty sure that this value is created somewhere else" :(

Reply:
Sorry, I don't understand.  The idea is the test needs to setup the environment to exercise the code.  Therefore, it needs to seed whatever tables it needs to test.  The `maintenance_zone`, as a query method, should not have side effects, such as creating a region.

Reply:
MiqRegion is the 2nd primordial class to be seeded.  If `MiqRegion.my_region` is nil, we've got much worse problems.

See [here](https://github.com/ManageIQ/manageiq/blob/3ae9ce6e535bdf80037a01e1dfd4d31ccb4ed3ff/lib/evm_database.rb#L6)

Reply:
Agreed I would rather rely on the order of the seeding (since we don't change that order often/ever) and maybe raise if that isn't true so we can catch it than try to handle every possible nil value everywhere.

Reply:
@jrafanie it can be true, but it's not excuse to raise exceptions on nil associations. I strongly disagree with this approach